yjit/src/asm/x86_64/mod.rs

```diff
@@ -117,6 +120,19 @@ pub enum X86Opnd
     IPRel(i32)
 }
 
+impl X86Opnd {
+    fn rex_needed(&self) -> bool {
+        match self {
```

Comment:
I think you can do the following to DRY things up:
```suggestion
        use Self::*;
        match self {
```
Not sure if `Self` works though